include/eigenpy/memory.hpp

```diff
@@ -1,13 +1,19 @@
 /*
  * Copyright 2014-2019, CNRS
- * Copyright 2018-2019, INRIA
+ * Copyright 2018-2022, INRIA
  */
 
 #ifndef __eigenpy_memory_hpp__
 #define __eigenpy_memory_hpp__
 
 #include "eigenpy/fwd.hpp"
 
+#ifndef Py_SET_SIZE
+#define EIGENPY_Py_SET_SIZE(object, size) Py_SIZE(object) = size
+#else
+#define EIGENPY_Py_SET_SIZE(object, size) Py_SET_SIZE(object, size)
+#endif
```

Comment:
```suggestion
#if PY_VERSION_HEX < 0x030900A4 && !defined(Py_SET_SIZE)
static inline void _Py_SET_SIZE(PyVarObject *ob, Py_ssize_t size)
{ ob->ob_size = size; }
#define Py_SET_SIZE(ob, size) _Py_SET_SIZE((PyVarObject*)(ob), size)
#endif
```

ref. https://docs.python.org/3.11/whatsnew/3.11.html#id6 ;